### ISSUE TITLE
Generic syntax error when 'module' declaration is missing 'exposing'

### ISSUE THREAD

**Issue by JohnBugner** [elm-lang/error-message-catalog#136]
Take the following code:

```
module ModuleA

import ModuleB

[... stuff ...]
```

The compiler gives the following error message:

```
I need whitespace, but got stuck on what looks like a new declaration. You are
either missing some stuff in the declaration above or just need to add some
spaces here:

3| import ModuleB
   ^
I am looking for one of the following things:

    whitespace
```

But what caused the error?: not the lack of whitespace!, but a missing 'exposing' clause after "module ModuleA". It's fixed by simply writing:

```
module ModuleA exposing (..)

import ModuleB

[... stuff ...]
```

This is an misleading error message. It mentions needing whitespace three times, but needing an 'exposing' clause zero times. A beginner could get stuck looking at whitespace, forgetting that they need an 'exposing' clause, because the module declaration already looks and feels right to them. (Especially because many other languages list the module declaration at the top of the file just like Elm, but then either don't let the user define which things are exported, or let the user leave out the export clause.)

The error message should explicitly mention needing an 'exposing' clause. Saying "You are
either missing some stuff in the declaration above" isn't good enough. 

(I admit that I don't know how easy it would be to change this error message, and what other cases it would affect; I don't know the inner workings of the Elm compiler. Perhaps this is an unrealistic request; I apologize if it is.)

**Comment by evancz** [comment]
I'm working on a faster parser, and it also gives nicer error messages. Here is what happens for your SSCCE:

![screen shot 2016-12-15 at 9 38 44 pm](https://cloud.githubusercontent.com/assets/1658058/21250961/14273872-c30f-11e6-9c3a-94ae1d28b62c.png)

**Comment by evancz** [comment]
I am finishing up a revamp of the parser, and I just tried this with my development build. It is now saying:

```
-- UNFINISHED MODULE DECLARATION -------------------------------------- temp.elm

I am parsing an `module` declaration, but I got stuck here:

1| module ModuleA
                 ^
Here are some examples of valid `module` declarations:

    module Main exposing (..)
    module Dict exposing (Dict, empty, get)

I generally recommend using an explicit exposing list. I can skip compiling a
bunch of files when the public interface of a module stays the same, so exposing
fewer values can help improve compile times!
```

This version should be available once Elm 0.19.1 is released!
